pkg/virtctl/adm/logverbosity/logverbosity.go

```diff
@@ -0,0 +1,304 @@
+	cmd.Flags().BoolVar(&virtControllerFlag, "virt-controller", virtControllerFlag, "Specify virt-controller log verbosity level.")
+	cmd.Flags().BoolVar(&virtHanderFlag, "virt-handler", virtHanderFlag, "Specify virt-handler log verbosity level.")
+	cmd.Flags().BoolVar(&virtLauncherFlag, "virt-launcher", virtLauncherFlag, "Specify virt-launcher log verbosity level.")
+	cmd.Flags().BoolVar(&virtOperatorFlag, "virt-operator", virtOperatorFlag, "Specify virt-operator log verbosity level.")
+	cmd.Flags().BoolVar(&allFlag, "all", allFlag, "Set all component log verbosity levels.")
+	cmd.Flags().BoolVar(&resetFlag, "reset", resetFlag, "Reset log verbosity level.")
+
+	// no required flag
+
+	cmd.SetUsageTemplate(templates.UsageTemplate())
+	return cmd
+}
+
+func usage() string {
+	usage := "  # Set a log verbosity level to a component:\n"
+	usage += "  {{ProgramName}} adm logVerbosity 3 --virt-api"
+	usage += "  # Set a log verbosity level to multiple components:\n"
+	usage += "  {{ProgramName}} adm logVerbosity 3 --virt-api  --virt-handler"
+	usage += "  # Set a log verbosity level to all components:\n"
+	usage += "  {{ProgramName}} adm logVerbosity 3 --all"
+	usage += "  # Reset the log verbosity level to all components:\n"
+	usage += "  {{ProgramName}} adm logVerbosity --reset"
+
+	return usage
+}
+
+func (c *Command) RunE(args []string) error {
+	/*
+	 * flag strength: reset > all > component
+	 */
+
+	var logLevel uint
+
+	// parse argument
+	if !resetFlag {
+		if len(args) == 0 {
+			return fmt.Errorf("expecting log verbosity level")
+		}
+		level, err := strconv.ParseUint(args[0], 10, 0)
+		if err != nil {
+			return err
+		}
+		logLevel = uint(level)
+
+		/*
+		 * user can specify logLevel 0
+		 * if we do not allow user to specify logLevel 0, need error handling here.
+		 */
+
+	}
+
+	// adm command always sees the KubeVirt resource of namespace="kubevirt", name="kubevirt"
+	const (
+		NAMESPACE = "kubevirt"
+		NAME      = "kubevirt"
+	)
+
+	virtClient, err := kubecli.GetKubevirtClientFromClientConfig(c.clientConfig)
+	if err != nil {
+		return fmt.Errorf("cannot obtain KubeVirt client: %v", err)
+	}
+
+	// get KubeVirt object
+	kv, err := virtClient.KubeVirt(NAMESPACE).Get(NAME, &k8smetav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	// create patch
+	var patch string
+
+	// get LogVerbosity strcut entries
+	lv := kv.Spec.Configuration.DeveloperConfiguration.LogVerbosity
+
+	if lv == nil {
+		// no LogVerbosity etnry yet
+
+		if resetFlag {
+
+			// do nothing, because there exist no LogVerbosity setting
+			return nil
+
+		} else if allFlag {
+
+			patch = fmt.Sprintf(`
+					[{"op": "add","path": "/spec/configuration/developerConfiguration/logVerbosity","value": {"virtHandler": %d, "virtAPI": %d, "virtController": %d, "virtLauncher": %d, "virtOperator": %d}}]
+				`, logLevel, logLevel, logLevel, logLevel, logLevel)
+
+		} else if virtAPIFlag || virtControllerFlag || virtHanderFlag || virtLauncherFlag || virtOperatorFlag {
```

Comment:
Please, replace this with a function, it makes it more readable especially if you are going to repeat this statement in multiple places

Reply:
Good catch. I was wondering if I could always assume that the namespace and name are "kubevirt". So I put these constants here temporarily.

I want to focus on the KubeVirt CR first, with the possibility of future expansion. So let me move the constants to the beginning for now.

I have an idea to get namespace and name like this.
```
namespace, _, err := clientConfig.Namespace()
kvList, err := virtClient.KubeVirt(namespace).List(&k8smetav1.ListOptions{})
kv := kvList.Items[0]
name = kv.Name
```
When I "make cluster-up" on my local environment, I failed to get the namespace and name by using the above method. I'll fix it and get the namespace and name, and remove the constants later.